### ISSUE TITLE
Crash on fork

### ISSUE THREAD

**Issue by aviramha** [metalbear-co/mirrord#1697]
Sometimes we see users getting this - 
```
thread '<unnamed>' panicked at 'failed to wake I/O driver: Os { code: 9, kind: Uncategorized, message: "Bad file descriptor" }', /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/io/mod.rs:255:27
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: tokio::runtime::scheduler::multi_thread::park::Unparker::unpark
   4: tokio::runtime::scheduler::multi_thread::worker::<impl tokio::runtime::task::Schedule for alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::handle::Handle>>::schedule
   5: tokio::runtime::task::waker::wake_by_val
   6: mirrord_layer::layer_start
   7: mirrord_layer::layer_pre_initialization
   8: std::panicking::try
   9: mirrord_layer::fork_detour
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
Which seems to stem from the fork logic.

**Comment by aviramha** [comment]
Might have been fixed by #1763

**Comment by aviramha** [comment]
Fixed by #1763
